integration_test/util_test.go

```diff
@@ -0,0 +1,361 @@
+
+	user1Mnemonics, err := os.ReadFile(filepath.Join(".", "test_keys", "user1.mnemonics"))
+	if err != nil {
+		panic(err)
+	}
+
+	userWallet1, err = NewEVMWalletInfo(string(user1Mnemonics))
+	if err != nil {
+		panic(err)
+	}
+
+	user2Mnemonics, err := os.ReadFile(filepath.Join(".", "test_keys", "user2.mnemonics"))
+	if err != nil {
+		panic(err)
+	}
+
+	userWallet2, err = NewEVMWalletInfo(string(user2Mnemonics))
+	if err != nil {
+		panic(err)
+	}
+
+	validator1Mnemonics, err := os.ReadFile(filepath.Join(".", "test_keys", "validator1.mnemonics"))
+	if err != nil {
+		panic(err)
+	}
+
+	validatorWallet1, err = NewEVMWalletInfo(string(validator1Mnemonics))
+	if err != nil {
+		panic(err)
+	}
+
+	return
+}
+
+func txCheck(txHash string) error {
+	var err error
+
+	for i := 0; i < 20; i++ {
+		txClient := txtypes.NewServiceClient(desc.GetConnectionWithContext(context.Background()))
+		_, err = txClient.GetTx(context.Background(), &txtypes.GetTxRequest{Hash: txHash})
+
+		if err == nil {
+			return nil
+		}
+
+		time.Sleep(time.Second / 2)
+	}
+
+	return err
+}
+
+func applyVoteTallyingProposal(conn *grpc.ClientConn, proposalContent govtype.Content, proposerWallet *WalletInfo, voters []*WalletInfo) error {
+	proposalId := uint64(0)
+
+	{
+		fmt.Println("Proposal apply")
+
+		proposalMsg, err := govtype.NewMsgSubmitProposal(
+			proposalContent,
+			sdktypes.NewCoins(sdktypes.NewCoin("axpla", sdktypes.NewInt(10000000))),
+			proposerWallet.ByteAddress,
+		)
+
+		if err != nil {
+			return err
+		}
+
+		feeAmt := sdktypes.NewDec(xplaProposalGasLimit).Mul(sdktypes.MustNewDecFromStr(xplaGasPrice))
+		fee := sdktypes.Coin{
+			Denom:  "axpla",
+			Amount: feeAmt.Ceil().RoundInt(),
+		}
+
+		txhash, err := proposerWallet.SendTx(ChainID, proposalMsg, fee, xplaProposalGasLimit, false)
+		if txhash != "" && err == nil {
+			fmt.Println("Tx sent:", txhash)
+		} else {
+			return err
+		}
+
+		err = txCheck(txhash)
+		if err == nil {
+			fmt.Println("Tx applied", txhash)
+		} else {
+			return err
+		}
+
+		queryClient := txtypes.NewServiceClient(conn)
+		resp, err := queryClient.GetTx(context.Background(), &txtypes.GetTxRequest{
+			Hash: txhash,
+		})
+
+		if err != nil {
+			return err
+		}
+
+	PROPOSAL_RAISED:
+		for _, val := range resp.TxResponse.Events {
+			for _, attr := range val.Attributes {
+				if string(attr.Key) == "proposal_id" {
+					proposalId, _ = strconv.ParseUint(string(attr.Value), 10, 64)
+					break PROPOSAL_RAISED
+				}
+			}
+		}
+
+		fmt.Println("Proposal is applied as ID", proposalId)
+	}
+
+	{
+		fmt.Println("Voting...")
+
+		wg := sync.WaitGroup{}
+
+		errChan := make(chan error)
+		successChan := make(chan bool)
+
+		for _, addr := range voters {
+			wg.Add(1)
+
+			go func(addr *WalletInfo) {
+				defer wg.Done()
+
+				voteMsg := govtype.NewMsgVote(addr.ByteAddress, proposalId, govtype.OptionYes)
+				feeAmt := sdktypes.NewDec(xplaGeneralGasLimit).Mul(sdktypes.MustNewDecFromStr(xplaGasPrice))
+				fee := sdktypes.Coin{
```

Comment:
`sdktypes.Coin{...}` can change `sdktypes.NewCoin()`

It can validate coin and simple to code

Reply:
Remain here.
- https://github.com/xpladev/xpla/pull/84/files#diff-9553717e46fb04a24106349dd56b086dc6c1daac928c607ff758f9612cbdd3b4R213
- https://github.com/xpladev/xpla/pull/84/files#diff-9553717e46fb04a24106349dd56b086dc6c1daac928c607ff758f9612cbdd3b4R270
-

Reply:
https://github.com/xpladev/xpla/pull/84/commits/7b8c6f60d746260c465ee1e1a31b374de41f4c23